jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java

```diff
@@ -112,6 +121,16 @@ public Group getGroup(final String name) {
     @Transactional
     public Group updateGroup(final UpdateGroupRequest anUpdateGroupRequest,
                              final User anUpdatingUser) {
+        Group orginalGroup = getGroup(anUpdateGroupRequest.getId());
+        try {
+            if (!orginalGroup.getName().equalsIgnoreCase(anUpdateGroupRequest.getNewName()) && null != groupPersistenceService.getGroup(anUpdateGroupRequest.getNewName())) {
+                String message = MessageFormat.format("Group Name already exists: {0}", anUpdateGroupRequest.getNewName());
+                LOGGER.error(message);
+                throw new EntityExistsException(message);
+            }
+        } catch (NotFoundException e) {
+            LOGGER.debug("No group name conflict, ignoring not found exception for creating group ", e);
+        }
         anUpdateGroupRequest.validate();
```

Comment:
If validate checks if the name is legal e.g. is null, has illegal characters, length is good etc then validate should go first before checking for duplicates